src/ClientServices/WindowsProcessLauncherClient.cpp

```diff
@@ -0,0 +1,182 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "ClientServices/WindowsProcessLauncherClient.h"
+
+#include <absl/container/flat_hash_map.h>
+#include <grpcpp/grpcpp.h>
+
+#include <chrono>
+#include <memory>
+
+#include "GrpcProtos/services.grpc.pb.h"
+#include "GrpcProtos/services.pb.h"
+#include "Introspection/Introspection.h"
+#include "OrbitBase/Logging.h"
+
+namespace orbit_client_services {
+namespace {
+
+using orbit_grpc_protos::LaunchProcessRequest;
+using orbit_grpc_protos::LaunchProcessResponse;
+using orbit_grpc_protos::ResumeProcessSuspendedAtEntryPointRequest;
+using orbit_grpc_protos::ResumeProcessSuspendedAtEntryPointResponse;
+using orbit_grpc_protos::SuspendProcessSpinningAtEntryPointRequest;
+using orbit_grpc_protos::SuspendProcessSpinningAtEntryPointResponse;
+
+class WindowsProcessLauncherClientImpl : public WindowsProcessLauncherClient {
+ public:
+  explicit WindowsProcessLauncherClientImpl(const std::shared_ptr<grpc::Channel>& channel)
+      : windows_process_launcher_service_(
+            orbit_grpc_protos::WindowsProcessLauncherService::NewStub(channel)) {}
+  virtual ~WindowsProcessLauncherClientImpl() = default;
+
+  ErrorMessageOr<orbit_grpc_protos::ProcessInfo> LaunchProcess(
+      const orbit_grpc_protos::ProcessToLaunch& process_to_launch) override;
+  ErrorMessageOr<void> SuspendProcessSpinningAtEntryPoint(uint32_t pid) override;
+  ErrorMessageOr<void> ResumeProcessSuspendedAtEntryPoint(uint32_t pid) override;
+
+  [[nodiscard]] bool IsProcessSpinningAtEntryPoint(uint32_t pid) override;
+  [[nodiscard]] bool IsProcessSuspendedAtEntryPoint(uint32_t pid) override;
+
+ private:
+  struct LaunchedProcess {
+    enum class State {
+      kInvalid,
+      kExecutingOrExited,
+      kSpinningAtEntryPoint,
+      kSuspendedAtEntryPoint
+    };
+    State state_ = State::kInvalid;
+    orbit_grpc_protos::ProcessInfo process_info_;
+  };
+
+  void UpdateLaunchedProcessState(uint32_t pid, LaunchedProcess::State new_state);
+
+  std::unique_ptr<orbit_grpc_protos::WindowsProcessLauncherService::Stub>
+      windows_process_launcher_service_;
+
+  absl::Mutex launched_processes_by_pid_mutex_;
+  absl::flat_hash_map<uint32_t, LaunchedProcess> launched_processes_by_pid_
+      ABSL_GUARDED_BY(launched_processes_by_pid_mutex_);
+};
+
+constexpr uint64_t kGrpcDefaultTimeoutMilliseconds = 3000;
```

Comment:
The obvious question is: why three seconds? I would add a little comment on the arbitrariness of this.